FixedPromyvions

Solarsurge opened the pull request:
*ALL Promyvion cutscenes now properly working according to mission progress.
*Fixed player dropoff after interacting with Shattered Telepoints during 2nd Promyvion
*Map acquisition now works properly in Holla and Mea

<!-- place 'x' mark between square [] brackets to affirm: -->
**_I affirm:_**
- [X] that I agree to Project Topaz's [Limited Contributor License Agreement](http://project-topaz.com/blob/release/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [X] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

Solarsurge committed:
FixedPromyvions

*ALL Promyvion cutscenes now properly working according to mission progress.
*Fixed player dropoff after interacting with Shattered Telepoints during 2nd Promyvion
*Map acquisition now works properly in Holla and Mea

Solarsurge committed:
Fixed Missing text in Zone IDs

ibm2431 commented on scripts/zones/Promyvion-Mea/npcs/qm1.lua:
```diff
@@ -0,0 +1,27 @@
+
+function onTrade(player, npc, trade)
+    if npcUtil.tradeHas(trade, 1722) and not player:hasKeyItem(tpz.ki.MAP_OF_PROMYVION_MEA) then
+        npcUtil.giveKeyItem(player, tpz.ki.MAP_OF_PROMYVION_MEA)
+        player:confirmTrade()
```
I know you're just using what's currently in these files, but while we're here, we might as well fix a retail inaccuracy~!

These should be starting an event where the screen flashes and there's text about blinding white lights and images being burned into your mind. This is event 49 for all three promyvions.

(And then give the KI and confirm the trade in onEventFinish for those events)

Solarsurge replied:
Fixed

ibm2431 commented on scripts/zones/Promyvion-Mea/npcs/qm1.lua:
```diff
@@ -0,0 +1,27 @@
+        npcUtil.giveKeyItem(player, tpz.ki.MAP_OF_PROMYVION_MEA)
+        player:confirmTrade()
+    else
+        player:messageSpecial(ID.text.NOTHING_HAPPENS)
```
Does this text display when you trade a bad item? Bad trades are usually just ignored.

Solarsurge replied:
"Nothing happens..." is displayed on a bad trade. To my knowledge, that's correct.

ibm2431 reviewed (changes requested):
You got close to a no-comment approval, but were sabotaged by previous scripters! 😉

Solarsurge committed:
Updated

*Added blinding light event on trade for map acquisition per request
*Fixed crash in Mea for missing text

Solarsurge committed:
Moved confirmTrade

ibm2431 reviewed (approved):
Thanks Solar!

TeoTwawki commented on scripts/zones/Promyvion-Dem/npcs/qm1.lua:
```diff
@@ -24,4 +25,7 @@ function onEventUpdate(player, csid, option)
 
 function onEventFinish(player, csid, option)
+    if csid == 49 then
+        npcUtil.giveKeyItem(player, tpz.ki.MAP_OF_PROMYVION_DEM)
```
Confirm trade should also be at event finish, not before it starts
